src/connectors/onchain_events/mod.rs

```diff
@@ -561,17 +626,36 @@ impl Subscriber {
     }
 
     async fn latest_block_on_chain(&mut self) -> Result<u64, SubscribeError> {
```

Comment:
Maybe it's worth having just 1 fn with the retry logic and feeding the actual rpc call in via an anon fn or closure. Might be annoying to deal with this in rust though, it's ok to leave as is.